Realm/ObjectServerTests/RLMAsymmetricSyncServerTests.mm

```diff
@@ -0,0 +1,316 @@
+                                 rootDirectory:self.clientDataRoot];
+        RLMSyncManager *syncManager = self.asymmetricSyncApp.syncManager;
+        syncManager.logLevel = RLMSyncLogLevelTrace;
+        syncManager.userAgent = self.name;
+    }
+    return _asymmetricSyncApp;
+}
+
+- (RLMUser *)userForSelector:(SEL)testSel {
+    return [self logInUserForCredentials:[self basicCredentialsWithName:NSStringFromSelector(testSel)
+                                                               register:YES
+                                                                    app:self.asymmetricSyncApp]
+                                     app:self.asymmetricSyncApp];
+}
+
+- (void)tearDown {
+    RLMUser *user = [self logInUserForCredentials:[RLMCredentials anonymousCredentials]
+                                              app:self.asymmetricSyncApp];
+    RLMMongoClient *client = [user mongoClientWithServiceName:@"mongodb1"];
+    RLMMongoDatabase *database = [client databaseWithName:@"test_data"];
+    RLMMongoCollection *collection = [database collectionWithName:@"PersonAsymmetric"];
+    [self cleanupRemoteDocuments:collection];
+    [super tearDown];
+}
+
+- (void)checkCountInMongo:(unsigned long)count {
+    RLMUser *user = [self logInUserForCredentials:[RLMCredentials anonymousCredentials]
+                                              app:self.asymmetricSyncApp];
+    RLMMongoClient *client = [user mongoClientWithServiceName:@"mongodb1"];
+    RLMMongoDatabase *database = [client databaseWithName:@"test_data"];
+    RLMMongoCollection *collection = [database collectionWithName:@"PersonAsymmetric"];
+    XCTestExpectation *findExpectation4 = [self expectationWithDescription:@"Should find documents"];
+    [collection findWhere:@{}
+               completion:^(NSArray<NSDictionary *> *documents, NSError *error) {
+        XCTAssertEqual(documents.count, count);
+        XCTAssertNil(error);
+        [findExpectation4 fulfill];
+    }];
+    [self waitForExpectationsWithTimeout:60.0 handler:nil];
+}
+
+- (void)testAsymmetricObjectSchema {
+    RLMUser *user = [self userForSelector:_cmd];
+    RLMRealmConfiguration *configuration = [user flexibleSyncConfiguration];
+    configuration.objectClasses = @[PersonAsymmetric.self];
+    RLMRealm *realm = [RLMRealm realmWithConfiguration:configuration error:nil];
+    XCTAssertTrue(realm.schema.objectSchema[0].isAsymmetric);
+}
+
+- (void)testUnsupportedAsymmetricLinkAsymmetricThrowsError {
+    RLMUser *user = [self userForSelector:_cmd];
+    RLMRealmConfiguration *configuration = [user flexibleSyncConfiguration];
+    configuration.objectClasses = @[UnsupportedLinkAsymmetric.self, PersonAsymmetric.self];
+    NSError *error;
+    RLMRealm *realm = [RLMRealm realmWithConfiguration:configuration error:&error];
+    XCTAssertNil(realm);
+    // This error comes from core, we are not even adding this objects to the server schema.
+    XCTAssertNotNil(error);
```

Comment:
We should verify that the error is something sensible and not just that there is an error in each of these tests.